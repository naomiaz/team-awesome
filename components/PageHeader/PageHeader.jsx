import React from 'react';
import PropTypes from 'prop-types';

import './PageHeader.scss';

const TimesheetHeader = ({ pageTitle, count, unit }) => (
  <section className="timesheet-header">
    <div className="timesheet-header__title-wrapper">
      <h2 className="timesheet-header__title">
        { pageTitle }
      </h2>
      <p className="timesheet-header__text">
        { count }
        &nbsp;
        { unit }
      </p>
    </div>

    <div className="timesheet-header__search">
      <form action="/action_page.php">
        <input className="timesheet-header__searchbar" type="search" id="site-search" name="search" placeholder="Search" aria-label="Search through site content" />
      </form>
    </div>
  </section>
);

TimesheetHeader.propTypes = {
  pageTitle: PropTypes.string.isRequired,
  count: PropTypes.number.isRequired,
  unit: PropTypes.string.isRequired
};

export default TimesheetHeader;
