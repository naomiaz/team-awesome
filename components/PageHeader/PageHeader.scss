@import "../../shared/styles/main";

.timesheet-header { display: none; }

@include tablet {
  .timesheet-header {
    align-items: center;
    box-shadow: 0 1px 0 0 $color-monochrome-2;
    display: flex;
    height: 70px;
    justify-content: space-between;
    padding: 0 30px;

    &__title,
    &__text {
      display: inline-block;
    }

    &__text {
      color: $color-monochrome-3;

      &::before {
        color: $color-monochrome-3;
        content: "|";
        font-weight: $font-weight-light;
        padding: 0 12px;
      }
    }

    &__searchbar {
      background: url("/static/icons/magnifying-glass.svg") top right no-repeat;
      border: 1px solid $color-monochrome-6;
      border-radius: $border-radius-primary;
      font-family: $font-primary;
      font-size: 14px;
      height: 36px;
      opacity: .6;
      padding: 0 15px;
      width: 240px;
    }
  }


} // tablet
