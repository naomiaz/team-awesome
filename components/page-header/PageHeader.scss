@import "../../shared/styles/main";

.page-header { display: none; }

@include tablet {
  .page-header {
    align-items: center;
    box-shadow: 0 1px 0 0 $color-monochrome-2;
    display: flex;
    height: 70px;
    justify-content: space-between;
    padding: 0 30px;

    &__title,
    &__text {
      display: inline-block;
    }

    &__text {
      color: $color-monochrome-3;

      &::before {
        color: $color-monochrome-3;
        content: "|";
        font-weight: $font-weight-light;
        padding: 0 12px;
      }
    }

    &__searchbar {
      background-color: transparent;
      border: 1px solid $color-monochrome-6;
      font-weight: 400;
      height: 36px;
      opacity: .6;
      padding: 0 32px 0 15px;
      width: 240px;

      &-btn {
        background-color: transparent;
        border: none;
        margin-left: -32px;
      }
    }
  }
}
