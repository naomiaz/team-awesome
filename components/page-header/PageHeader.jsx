import React from 'react';
import PropTypes from 'prop-types';

import './PageHeader.scss';

const PageHeader = ({ pageTitle, unitCount, unit }) => (
  <section className="page-header">
    <div className="page-header__title-wrapper">
      <h2 className="page-header__title">
        { pageTitle }
      </h2>
      <p className="page-header__text">
        { unitCount }
        &nbsp;
        { unit }
      </p>
    </div>

    <div className="page-header__search">
      <form action="/search_page.php">
        <input
          aria-label="Search through site content"
          className="page-header__searchbar"
          id="site-search"
          name="search"
          placeholder="Search"
          type="search"
        />
        <button
          className="page-header__searchbar-btn"
          type="button"
        >
          <img
            alt=""
            src="../../static/icons/magnifying-glass.svg"
          />
        </button>
      </form>
    </div>
  </section>
);

PageHeader.propTypes = {
  pageTitle: PropTypes.string.isRequired,
  unitCount: PropTypes.number.isRequired,
  unit: PropTypes.string.isRequired
};

export default PageHeader;
