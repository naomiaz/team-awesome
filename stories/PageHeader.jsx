import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import PageHeader from '../shared/components/page-header/PageHeader';

storiesOf('PageHeader', module)
  .add('without filter', () => (
    <PageHeader
      pageTitle="Time Entries"
      unitCount={2}
      unitSingular="Entry"
      unitPlural="Entries"
    />
  ))
  .add('with filter', () => (
    <PageHeader
      pageTitle="Time Entries"
      selectBox={[{
        onChange: action('changed filter value'),
        options: [
          { title: 'All clients:', value: '' },
          { title: 'Albert Heijn', value: 'Albert Heijn' },
          { title: 'Hike One', value: 'Hike One' },
          { title: 'Port of Rotterdam', value: 'Port of Rotterdam' }
        ],
        selectedValue: ''
      }]}
      unitCount={1}
      unitSingular="Entry"
      unitPlural="Entries"
    />
  ));
