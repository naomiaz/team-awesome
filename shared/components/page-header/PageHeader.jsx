import React from 'react';
import PropTypes from 'prop-types';
import SelectBox from '../select-box/SelectBox';

import './page-header.scss';

const PageHeader = ({
  pageTitle, selectBox, unitCount, unitSingular, unitPlural
}) => (
  <section className="page-header">
    <h1 className="page-header__title">
      { pageTitle }
      <span className="page-header__text">
        {`${unitCount} ${unitCount === 1 ? unitSingular : unitPlural}`}
      </span>
    </h1>

    <div className="page-header__button-wrapper">
      {selectBox.map((currentSelectBox, index) => (
        <SelectBox
          key={index} // eslint-disable-line
          {...currentSelectBox}
        />
      ))
      }

      <form className="page-header__search">
        <input
          className="page-header__search-field"
          name="search"
          placeholder="Search"
          type="search"
        />
        <button
          className="page-header__button-search"
          type="submit"
        />
      </form>
    </div>
  </section>
);

PageHeader.defaultProps = {
  selectBox: []
};

PageHeader.propTypes = {
  pageTitle: PropTypes.string.isRequired,
  selectBox: PropTypes.arrayOf(
    PropTypes.shape({
      className: PropTypes.string,
      name: PropTypes.string,
      onChange: PropTypes.func.isRequired,
      options: PropTypes.shape({
        title: PropTypes.string.isRequired,
        value: PropTypes.string.isRequired
      }).isRequired,
      selectedValue: PropTypes.string.isRequired
    })
  ),
  unitCount: PropTypes.number.isRequired,
  unitSingular: PropTypes.string.isRequired,
  unitPlural: PropTypes.string.isRequired
};

export default PageHeader;
