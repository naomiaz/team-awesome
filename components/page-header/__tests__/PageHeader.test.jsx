import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PageHeader from '../PageHeader';

it('renders correctly', () => {
  const wrapper = shallow(
    <PageHeader
      pageTitle="Timesheet"
      unitCount={12}
      unit="entries"
    />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
