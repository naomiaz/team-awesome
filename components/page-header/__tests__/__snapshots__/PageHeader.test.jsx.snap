// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="page-header"
>
  <div
    className="page-header__title-wrapper"
  >
    <h2
      className="page-header__title"
    >
      Timesheet
    </h2>
    <p
      className="page-header__text"
    >
      12
      Â 
      entries
    </p>
  </div>
  <div
    className="page-header__search"
  >
    <form
      action="/"
    >
      <input
        aria-label="Search through site content"
        className="input-field page-header__searchbar"
        id="site-search"
        name="search"
        placeholder="Search"
        type="search"
      />
      <button
        className="page-header__searchbar-btn"
        type="button"
      >
        <img
          alt=""
          src="../../static/icons/magnifying-glass.svg"
        />
      </button>
    </form>
  </div>
</section>
`;
